pyphare/pyphare/pharesee/hierarchy.py

```diff
@@ -1362,11 +1786,17 @@ def hierarchy_fromh5(h5_filename, time, hier, silent=True):
 
                     patch_levels[ilvl] = PatchLevel(ilvl, patches[ilvl])
 
-        diag_hier = PatchHierarchy(
-            patch_levels, domain_box, refinement_ratio, t, data_file
-        )
+        if num_of_components == 1:
```

Comment:
I don't think we want to hard-code that we want a Scalar or Vector or Tensor or anything in that function that should just return a `PatchHierarchy`.

Indeed it is not because the h5 has 3 datasets that they necessarily are components of a vectorfield.. could be Bx, N, P.

We should keep these high level mathematical types of `PatchHierarchy`(scalar, vectors, etc). in physical models such as `run`. 




```python
run.GetB() # should return a VectorField
run.GetNi() # should return a ScalarField
run.GetPressure() # should return a TensorField
```